File: web/src/templates/domain/TemplateRepository.ts
import type { AssessmentTemplate, ImportedTemplate, TemplateDetailDto } from './AssessmentTemplate'

export interface TemplateRepository {
  getAll(): Promise<AssessmentTemplate[]>
  getById(id: number): Promise<TemplateDetailDto>
  importTemplate(name: string, file: File): Promise<ImportedTemplate>
  deleteTemplate(id: number): Promise<void>
}

File: backend/Assessments/src/main/kotlin/templates/domain/TemplateRepository.kt
package org.example.templates.domain

interface TemplateRepository {
    suspend fun importTemplate(name: String, parsed: ParsedTemplate): ImportResult
    suspend fun getAll(): List<AssessmentTemplate>
    suspend fun getById(id: Int): TemplateDetailDto?
    suspend fun delete(id: Int): Boolean
}

sealed class ImportResult {
    data class Success(val templateId: Int, val itemCount: Int) : ImportResult()
}
